src/app/routes/connector-ui/policy-definition-page/new-policy-dialog/new-policy-dialog-form.ts

```diff
@@ -29,41 +30,26 @@ export class NewPolicyDialogForm {
 
   constructor(private formBuilder: FormBuilder) {}
 
-  buildFormGroup(): FormGroup<NewPolicyDialogFormModel> {
-    const newPolicyFormGroup: FormGroup<NewPolicyDialogFormModel> =
-      this.formBuilder.nonNullable.group({
-        id: ['', [Validators.required, noWhitespaceValidator]],
-        policyType: [
-          'Connector-Restricted-Usage' as PolicyType,
-          Validators.required,
-        ],
-        range: this.formBuilder.group(
-          {
-            start: null as Date | null,
-            end: null as Date | null,
-          },
-          {validators: dateRangeRequired},
-        ),
-        connectorId: ['', Validators.required],
-      });
-
-    switchDisabledControls<NewPolicyDialogFormValue>(
-      newPolicyFormGroup,
-      (value) => {
-        const timePeriodRestricted =
-          value.policyType === 'Time-Period-Restricted';
-        const connecterRestrictedUsage =
-          value.policyType === 'Connector-Restricted-Usage';
-
-        return {
-          id: true,
-          policyType: true,
-          range: timePeriodRestricted,
-          connectorId: connecterRestrictedUsage,
-        };
-      },
-    );
+  buildFormGroup(): FormGroup {
+    return this.formBuilder.nonNullable.group({
+      policyDefinitionId: ['', [Validators.required, noWhitespaceValidator]],
+      policyType: [
+        'Connector-Restricted-Usage' as PolicyType,
+        Validators.required,
+      ],
+      uiPolicyDto: this.formBuilder.group({
+        constraints: this.formBuilder.array([this.buildConstraintForm()]),
+      }),
+    });
+  }
 
-    return newPolicyFormGroup;
+  buildConstraintForm(): FormGroup {
+    return this.formBuilder.group({
+      policyDefinitionId: true,
+      policyType: true,
+      left: ['', Validators.required],
+      operator: ['', Validators.required],
+      right: this.formBuilder.group({}),
```

Comment:
this should represent the DOM of the form, e.g type selector, the different inputs that will be only visible depending on the type, etc.